public/app/plugins/panel/table/TablePanel.tsx

```diff
@@ -68,13 +76,17 @@ export class TablePanel extends Component<Props> {
   onCellFilterAdded = (filter: FilterItem) => {
     const { key, value, operator } = filter;
     const panelModel = getDashboardSrv().getCurrent()?.getPanelById(this.props.id);
-    const datasource = panelModel?.datasource;
 
-    if (!datasource) {
+    // The datasource ref from the panel model can be null/undefined if a panel uses a default datasource
+    // so we need to resolve it to the real final datasource.
```

Comment:
i think it's worth clarifying that `getInstanceSettings` will return the default datasource instance settings if the argument (in this case `panelModel?.datasource`) is undefined